pkg/workflow/dataimpl.go

```diff
@@ -134,3 +251,56 @@ func (d *DataImpl) GetErrorList() []snyk_errors.Error {
 func (d *DataImpl) AddError(err snyk_errors.Error) {
 	d.errors = append(d.errors, err)
 }
+
+func (d *DataImpl) SetConfiguration(config configuration.Configuration) {
+	d.configuration = config
+}
+
+func (d *DataImpl) GetConfiguration() configuration.Configuration {
+	return d.configuration
+}
+
+func setPayloadLocation(id Identifier, config configuration.Configuration, payload interface{}, logger *zerolog.Logger) Location {
+	payloadLocation := Location{
+		Path: "",
+		Type: InMemory,
+	}
+
+	logger.Debug().Msg("checking if payload is []byte")
+	bytes, ok := payload.([]byte)
+	if !ok {
+		return payloadLocation
+	}
+
+	payloadSize := len(bytes)
+	fileSizeThreshold := config.GetInt(configuration.IN_MEMORY_THRESHOLD_BYTES)
+	logger.Debug().Msgf("payload is []byte, comparing payload size (%d bytes) to threshold (%d bytes)", payloadSize, fileSizeThreshold)
+
+	if payloadSize <= fileSizeThreshold {
+		logger.Debug().Msg("payload is lower than threshold, keeping it in memory")
+		return payloadLocation
+	}
+
+	logger.Debug().Msg("payload is larger than threshold, writing it to disk")
+	return writePayloadToDisk(id, config, bytes, logger)
+}
+
+func writePayloadToDisk(id Identifier, config configuration.Configuration, data []byte, logger *zerolog.Logger) Location {
+	tempDir := config.GetString(configuration.TEMP_DIR_PATH)
+	filepath := fmt.Sprintf("%s/workflow.%s.%d", tempDir, id.Path, time.Now().Unix())
```

Comment:
Suggestion: Better use [stdlib functionality](https://pkg.go.dev/os#CreateTemp) this is safer than using a unix timestamp.